### ISSUE TITLE
Getting the config when the plugin is initialized

### ISSUE THREAD

**Issue by francois2metz** [turbot/steampipe-plugin-sdk#174]
For my [airtable plugin](https://github.com/francois2metz/steampipe-plugin-airtable) I need the connection configuration to dynamically create the tables. This doesn't seems to be possible right now. 
I haven't digged in the code to see if it's hard to implement or not. 
If you have any hints on how I could do that, it would be nice.

**Comment by e-gineer** [comment]
Thanks @francois2metz for opening this issue from our prior discussions in Slack. Definitely excited to see what's possible once we can be more dynamic in the table definitions phase!

@kaidaguerre has started thinking about this and mocking some approaches, so I'll let him expand below.

**Comment by kaidaguerre** [comment]
Hi @francois2metz, thanks for raising this issue. 

There is a bootstrap ordering issue which makes it hard to provide the parsed connection config at plugin startup time. 

So instead what I am proposing to do is provide a callback function which can optionally be used to create the tables. The connection config would be available within this function so dynamic table   

So an example usage may look like this: 
```

func Plugin(ctx context.Context) *plugin.Plugin {
	return &plugin.Plugin{
		Name: pluginName,
		ConnectionConfigSchema: &plugin.ConnectionConfigSchema{
			NewInstance: ConfigInstance,
			Schema:      ConfigSchema,
		},

		// set create tables callback function 
		TableMapFunc: createTables,
	}
}

// table creation factory function
func createTables(p *plugin.Plugin) (map[string]*plugin.Table, error) {
	// get connection config from plugin
	connectionConfig := GetConfig(p.Connection)
	
	var tableMap = make(map[string]*plugin.Table)
	
	// TODO dynamically build tables using connectionConfig
	
	
	return tableMap, nil
}

```

**Comment by francois2metz** [comment]
That looks good!

**Comment by francois2metz** [comment]
@kaidaguerre Awesome work! I used the new version on https://github.com/francois2metz/steampipe-plugin-airtable/blob/main/airtable/plugin.go. 
For some reasons, when using the callback function, the descriptions of the tables and columns is not working.
```
> .inspect airtable
+----------------------+-------------+
| table                | description |
+----------------------+-------------+
| airtable_acquis      |             |
| airtable_competences |             |
| airtable_domaines    |             |
| airtable_epreuves    |             |
| airtable_referentiel |             |
| airtable_tubes       |             |
+----------------------+-------------+

To get information about the columns in a table, run .inspect {c
onnection}.{table}

> .inspect airtable_acquis
+--------+-------+-------------+
| column | type  | description |
+--------+-------+-------------+
| fields | jsonb |             |
| id     | text  |             |
+--------+-------+-------------+
```

**Comment by kaidaguerre** [comment]
@francois2metz thanks for spotting this - it is actually a small bug in the Steampipe CLI - when retrieving the plugin schema in order to populate the `inspect` data, the connection config was not being passed to the plugin. This resulted in the descriptions being omitted for dynamic tables. https://github.com/turbot/steampipe/issues/932

I have fixed this and it will be included in the forthcoming `0.8.3` patch, which will be released this week
